### ISSUE TITLE
The child tables does not inherit the same ownership as the parent table

### ISSUE THREAD

**Issue by hrawulwa** [pgpartman/pg_partman#562]
Hi,
After I created the partitioned table as postgres user, changed ownership of the table to application user.
Then created the partitions using partman.create_parent function as postgres user.
However, I see that child tables are assigned owner as postgres user, not the one I explicitly assigned to the main partitioned table.
I thought the ownership is inherited by child tables as per the documentation.
How do I ensure that all the child tables get the same owner as the main table?
Please advise.

Thanks
Hari

**Comment by keithf4** [comment]
The owner of a table is whoever creates it, so if you're running maintenance as the postgres super user, that is who will own them. If you run maintenance as the owner of the parent, then the children will have the same owner. 

Also by default, privileges of the parent are not inherited to the children, so you can only access the table through the parent. 

If you want to be able to run maintenance as a role that is not the table but keep the privileges, set the `inherit_privileges` option in the part_config table. Either that, or you can do what the documentation recommends and have a dedicated role that owns and manages partitions and have that role run maintenance.

Let me know if that works.

**Comment by keithf4** [comment]
Did you have any further questions on this issue?

**Comment by hrawulwa** [comment]
As per your recommendation, I used inherit_privileges option in part_config table.
Running the maintenance as postgres user, did create new partitions with the owner that was assigned to the parent table.
I was initially worried (concern raised by application team), since the data is inserted by application as the application user, there were questions if the partitions would accept the data, as the ownership is not the same as the table. But we have found out, irrespective of ownership of child tables, application can insert data as long as it has permission to write to the parent table. 
Using inherit_privileges is just convenience for application team, incase they have access the child table separately, which is usually not required.
Thanks for your help in providing clarifications.
